File: 121_Best_Time_to_Buy_and_Sell_Stock/121.py
from typing import List, Dict
from queue import Queue
from collections import deque, Counter
import math
from heapq import heappush, heappushpop

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        cost, profit = float("inf"), 0
        for price in prices:
            cost = min(cost, price)
            profit = max(profit, price - cost)
        return profit

File: 121_Best_Time_to_Buy_and_Sell_Stock/121.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int profit = 0;
        for (int price : prices) {
            if (price < minPrice) minPrice = price;
            if (price - minPrice > profit) profit = price - minPrice;
        }
        return profit;
    }
}